### ISSUE TITLE
[Content organization] Content organization and/or categorization isn't clear or actionable.  (02.06.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#61158]
#### VFS team name
Authenticated Experience

#### VFS product name
My VA

#### VFS feature name
Notification design system component

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Content organization and/or categorization isn't clear or actionable. 

### Issue Details
When multiple debt notifications are present, each notification has an identical heading. For screen reader users navigating the page by heading, it's difficult to distinguish the notifications from each other.

### VA.gov Experience Standard
[Category Number 02, Issue Number 06](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 3.1

### Platform Recommendation
This is something that I also flagged at the design intent for the Notification Center (#60486), and seeing the notifications here in staging has me reconsidering the feedback I gave there.

In that feedback, I suggested using `aria-describedby` to associate the date with the heading, so that screen readers would announce them together when announcing the heading, eg. "You have new debt Wednesday, June 28, 2023." I think that could be a reasonable solution for this particular notification in this context, but I'm less certain about how to handle this in a component lots of teams may try to implement down the line in unexpected ways.

As implemented here for debt, the notification body is just the date that the debt was posted. But in Storybook and in the design system guidance, there's no restriction on what the notification body can consist of besides being concise. With the body unrestricted, we likely don't want to hard-code a decision to use `aria-describedby` into the component --- and making a toggle to turn that `aria-describedby` relationship on or off seems over-complicated and would require a lot of explaining in the design system guidance.

It might be worth exploring building some extra structure into the component, like a required `dateTime` property that ensures the date is always included in the component, and thus always available as a way to make the headings relatively unique.

That could also let you include the date in the heading element itself while styling it differently. I'm thinking something like an eyebrow pattern, eg. `<h3><span class="eyebrow-date">June 28, 2023</span><span class="heading-text">You have new debt</span></h3>`. It would be announced as a single heading without any ARIA required, but could be styled so that the date and heading text are visually distinct.

But that approach might also mean some notifications have an empty body and just consist of date + heading text + action link. I don't know if that's desirable! Regardless, it would probably require some design work. I'm sorry to send you back down that path so late in the process.

Summarizing all of the above: There needs to be some way to make notification headings relatively unique. Doing something with `aria-describedby` would work for the notification you've built for debts, but that might not be the best or most robust solution moving forward. I'm very happy to discuss this further and help in any way I can.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by aagosto90** [comment]
Hey @briandeconinck ! We've had a few of these aria label tickets for me to address lately so it's been a good opportunity for me to brush up on my a11y knowledge. Could you explain to me why we should use `aria-describedby` for something like this as opposed to `aria-labelledby`? It seems like since the date text that we want to include in the screen reader announcement with the heading of the notification is visible on the page, we could use `aria-labelledby` but I'm still learning the nuances of aria labels so let me know if that's not the best solution. Thanks so much for your guidance on this!

**Comment by ajakabcin** [comment]
We'll address this in #61388 and #61399

**Comment by briandeconinck** [comment]
@aagosto90 Sorry for getting to this so late. Josh was exactly right in the [Slack thread](https://dsva.slack.com/archives/C8E985R32/p1688741391811409) -- `aria-labelledby` would prompt a screen reader to only read the element referenced by the `labelledby`. 

Borrowing from one of MDN's examples, here's the quick-and-dirty breakdown of what screen readers announce with common ARIA text:

```
<button>Red</button>
```
Screen reader announces: `Red`.

```
<button aria-label="Blue">Red</button>
```
Screen reader announces: `Blue`.

```
<button aria-labelledby="color">Red</button>
<span id="color">Blue</span>
```
Screen reader announces: `Blue`.

```
<button aria-describedby="color">Red</button>
<span id="color">Blue</span>
```
Screen reader announces: `Red Blue`.

But both `aria-labelledby` and `aria-describedby` allow you to reference multiple other elements to combine them into a longer string. So you can also do something like:

```
<button aria-labelledby="color1 color2">Red</button>
<span id="color1">Blue</span>
<span id="color2">Yellow</span>
```
Screen reader announces: `Blue Yellow`.

And:
```
<button aria-describedby="color1 color2">Red</button>
<span id="color1">Blue</span>
<span id="color2">Yellow</span>
```
Screen reader announces: `Red Blue Yellow`.

And it's valid (but unusual) to reference the element itself in the `aria-labelledby`, so something like:
```
<button id="my-btn" aria-labelledby="color1 my-btn color2">Red</button>
<span id="color1">Blue</span>
<span id="color2">Yellow</span>
```
Screen reader announces: `Blue Red Yellow`.

Maybe you'd do that if there's a specific order in which things need to be read that's not possible using `aria-describedby`. But we should really talk before you do something like that! 🙂
